runners/flink/src/main/java/org/apache/beam/runners/flink/translation/utils/FlinkPipelineTranslatorUtils.java

```diff
@@ -19,34 +19,15 @@
 
 import java.io.IOException;
 import org.apache.beam.model.pipeline.v1.RunnerApi;
-import org.apache.beam.runners.core.construction.RehydratedComponents;
-import org.apache.beam.runners.core.construction.WindowingStrategyTranslation;
 import org.apache.beam.runners.core.construction.graph.PipelineNode;
 import org.apache.beam.runners.fnexecution.wire.WireCoders;
 import org.apache.beam.sdk.coders.Coder;
 import org.apache.beam.sdk.util.WindowedValue;
-import org.apache.beam.sdk.values.WindowingStrategy;
-import org.apache.beam.vendor.grpc.v1p13p1.com.google.protobuf.InvalidProtocolBufferException;
-import org.apache.beam.vendor.guava.v20_0.com.google.common.collect.BiMap;
-import org.apache.beam.vendor.guava.v20_0.com.google.common.collect.ImmutableBiMap;
-import org.apache.beam.vendor.guava.v20_0.com.google.common.collect.Sets;
 
 /** Utilities for pipeline translation. */
 public final class FlinkPipelineTranslatorUtils {
```

Comment:
We can move this complete class to common location.

Reply:
Moving this last function into `core-construction` creates a circular dependency with `beam-runners-java-fn-execution`.

Reply:
instantiateCoder is only used in FlinkBatchPortablePipelineTranslator and FlinkStreamingPortablePipelineTranslator. So moving it should not create a cycle.

Reply:
I have moved the files into the `java-fn-execution/translation` directory I created, which is a more logical place and resolves this conflict.